{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "input_file_directory = '/Users/kja11/OneDrive - Háskóli Íslands/PhD ATG7//0 in_silico/Python/1)data_input/'\n",
    "output_file_directory = '/Users/kja11/OneDrive - Háskóli Íslands/PhD ATG7/0 in_silico/Python/3)output/'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "ifhgKXirU9Ds"
   },
   "source": [
    "# I] Import"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "id": "TLQwwG5VDPFA"
   },
   "outputs": [],
   "source": [
    "import pandas"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# II] ATG7 isoform expression"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(16179, 20)\n",
      "Wall time: 79.8 ms\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>Sample_Type</th>\n",
       "      <th>Primary_Site</th>\n",
       "      <th>ATG7</th>\n",
       "      <th>ATG7_1</th>\n",
       "      <th>ATG7_2</th>\n",
       "      <th>ATG7_3</th>\n",
       "      <th>ATG7_total</th>\n",
       "      <th>ATG7_1tpm</th>\n",
       "      <th>ATG7_2tpm</th>\n",
       "      <th>ATG7_3tpm</th>\n",
       "      <th>log2_p1_ATG7_total</th>\n",
       "      <th>log2_p1_ATG7_1</th>\n",
       "      <th>log2_p1_ATG7_2</th>\n",
       "      <th>ATG7_prot_tot</th>\n",
       "      <th>ATG7_1perc</th>\n",
       "      <th>ATG7_2perc</th>\n",
       "      <th>ATG7_3perc</th>\n",
       "      <th>ATG7_1norm</th>\n",
       "      <th>ATG7_2norm</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GTEX-ZTTD-0326-SM-57WFW</td>\n",
       "      <td>Normal Tissue</td>\n",
       "      <td>Muscle</td>\n",
       "      <td>5.042</td>\n",
       "      <td>3.860</td>\n",
       "      <td>2.390</td>\n",
       "      <td>0.2642</td>\n",
       "      <td>32.944283</td>\n",
       "      <td>14.519306</td>\n",
       "      <td>5.240574</td>\n",
       "      <td>1.199970</td>\n",
       "      <td>5.085097</td>\n",
       "      <td>3.955992</td>\n",
       "      <td>2.641679</td>\n",
       "      <td>20.95985</td>\n",
       "      <td>69.271996</td>\n",
       "      <td>25.002916</td>\n",
       "      <td>5.725088</td>\n",
       "      <td>0.440723</td>\n",
       "      <td>0.159074</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GTEX-PX3G-1626-SM-2S1PT</td>\n",
       "      <td>Normal Tissue</td>\n",
       "      <td>Muscle</td>\n",
       "      <td>4.252</td>\n",
       "      <td>2.322</td>\n",
       "      <td>1.956</td>\n",
       "      <td>1.1320</td>\n",
       "      <td>19.052710</td>\n",
       "      <td>4.999249</td>\n",
       "      <td>3.878848</td>\n",
       "      <td>2.190624</td>\n",
       "      <td>4.325725</td>\n",
       "      <td>2.584782</td>\n",
       "      <td>2.286540</td>\n",
       "      <td>11.06872</td>\n",
       "      <td>45.165557</td>\n",
       "      <td>35.043325</td>\n",
       "      <td>19.791118</td>\n",
       "      <td>0.262390</td>\n",
       "      <td>0.203585</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    sample    Sample_Type Primary_Site   ATG7  ATG7_1  ATG7_2  \\\n",
       "0  GTEX-ZTTD-0326-SM-57WFW  Normal Tissue       Muscle  5.042   3.860   2.390   \n",
       "1  GTEX-PX3G-1626-SM-2S1PT  Normal Tissue       Muscle  4.252   2.322   1.956   \n",
       "\n",
       "   ATG7_3  ATG7_total  ATG7_1tpm  ATG7_2tpm  ATG7_3tpm  log2_p1_ATG7_total  \\\n",
       "0  0.2642   32.944283  14.519306   5.240574   1.199970            5.085097   \n",
       "1  1.1320   19.052710   4.999249   3.878848   2.190624            4.325725   \n",
       "\n",
       "   log2_p1_ATG7_1  log2_p1_ATG7_2  ATG7_prot_tot  ATG7_1perc  ATG7_2perc  \\\n",
       "0        3.955992        2.641679       20.95985   69.271996   25.002916   \n",
       "1        2.584782        2.286540       11.06872   45.165557   35.043325   \n",
       "\n",
       "   ATG7_3perc  ATG7_1norm  ATG7_2norm  \n",
       "0    5.725088    0.440723    0.159074  \n",
       "1   19.791118    0.262390    0.203585  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%%time\n",
    "# DL data ATG7 isoform expression\n",
    "path = output_file_directory + \"/ATG7/TCGA_GTEX/all_tissues/dataframes_for_input/df_ATG7_NormPrim_all.csv\"\n",
    "df_ATG7_NormPrim= pandas.read_csv(path, sep = \",\")\n",
    "\n",
    "#remove Survival information\n",
    "df_ATG7_NormPrim.drop(df_ATG7_NormPrim.columns[[3, 4]], axis = 1, inplace = True)\n",
    "\n",
    "print(df_ATG7_NormPrim.shape)\n",
    "df_ATG7_NormPrim.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7401\n",
      "8778\n"
     ]
    }
   ],
   "source": [
    "#split to have Normal (GTEX) and Primary (TCGA)\n",
    "df_ATG7_Normal = df_ATG7_NormPrim[df_ATG7_NormPrim['sample'].str.startswith('GTEX')]\n",
    "print(len(df_ATG7_Normal))\n",
    "df_ATG7_Primary = df_ATG7_NormPrim[df_ATG7_NormPrim['sample'].str.startswith('TCGA')]\n",
    "print(len(df_ATG7_Primary))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "fbjL51C8RltI"
   },
   "source": [
    "## A) ATG7 isoforms + Whole gene expression (ensembl)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(167, 19383)\n",
      "(178, 19383)\n",
      "Wall time: 5.76 s\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>ENSG00000198888</th>\n",
       "      <th>ENSG00000198763</th>\n",
       "      <th>ENSG00000198804</th>\n",
       "      <th>ENSG00000198712</th>\n",
       "      <th>ENSG00000228253</th>\n",
       "      <th>ENSG00000198899</th>\n",
       "      <th>ENSG00000198938</th>\n",
       "      <th>ENSG00000198840</th>\n",
       "      <th>ENSG00000212907</th>\n",
       "      <th>...</th>\n",
       "      <th>ENSG00000160678</th>\n",
       "      <th>ENSG00000160679</th>\n",
       "      <th>ENSG00000143553</th>\n",
       "      <th>ENSG00000214193</th>\n",
       "      <th>ENSG00000196182</th>\n",
       "      <th>ENSG00000181817</th>\n",
       "      <th>ENSG00000116885</th>\n",
       "      <th>ENSG00000116898</th>\n",
       "      <th>ENSG00000119535</th>\n",
       "      <th>ENSG00000142694</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TCGA-2J-AAB1-01</td>\n",
       "      <td>12.5979</td>\n",
       "      <td>12.7660</td>\n",
       "      <td>13.9326</td>\n",
       "      <td>13.5892</td>\n",
       "      <td>13.4221</td>\n",
       "      <td>13.6399</td>\n",
       "      <td>13.5917</td>\n",
       "      <td>13.4003</td>\n",
       "      <td>12.4744</td>\n",
       "      <td>...</td>\n",
       "      <td>5.2173</td>\n",
       "      <td>5.8480</td>\n",
       "      <td>4.8861</td>\n",
       "      <td>3.9127</td>\n",
       "      <td>4.1628</td>\n",
       "      <td>5.1575</td>\n",
       "      <td>2.9338</td>\n",
       "      <td>5.9086</td>\n",
       "      <td>2.3423</td>\n",
       "      <td>4.7549</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TCGA-2J-AAB4-01</td>\n",
       "      <td>11.9508</td>\n",
       "      <td>12.2991</td>\n",
       "      <td>13.7032</td>\n",
       "      <td>13.9784</td>\n",
       "      <td>13.5229</td>\n",
       "      <td>13.5828</td>\n",
       "      <td>13.9554</td>\n",
       "      <td>12.9003</td>\n",
       "      <td>12.4813</td>\n",
       "      <td>...</td>\n",
       "      <td>3.6703</td>\n",
       "      <td>5.9189</td>\n",
       "      <td>4.9275</td>\n",
       "      <td>2.8381</td>\n",
       "      <td>4.5675</td>\n",
       "      <td>4.7793</td>\n",
       "      <td>2.7615</td>\n",
       "      <td>5.8590</td>\n",
       "      <td>2.7139</td>\n",
       "      <td>4.4054</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TCGA-2J-AAB6-01</td>\n",
       "      <td>11.0837</td>\n",
       "      <td>11.8905</td>\n",
       "      <td>12.5723</td>\n",
       "      <td>12.9621</td>\n",
       "      <td>12.3837</td>\n",
       "      <td>12.5712</td>\n",
       "      <td>13.6005</td>\n",
       "      <td>11.8531</td>\n",
       "      <td>11.6337</td>\n",
       "      <td>...</td>\n",
       "      <td>1.9149</td>\n",
       "      <td>5.5756</td>\n",
       "      <td>5.1728</td>\n",
       "      <td>3.5174</td>\n",
       "      <td>4.1652</td>\n",
       "      <td>5.5085</td>\n",
       "      <td>2.2663</td>\n",
       "      <td>7.0333</td>\n",
       "      <td>2.4412</td>\n",
       "      <td>5.2377</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TCGA-2J-AAB8-01</td>\n",
       "      <td>10.7231</td>\n",
       "      <td>12.7880</td>\n",
       "      <td>12.1531</td>\n",
       "      <td>13.7012</td>\n",
       "      <td>13.7530</td>\n",
       "      <td>13.4922</td>\n",
       "      <td>13.7765</td>\n",
       "      <td>13.2663</td>\n",
       "      <td>12.7046</td>\n",
       "      <td>...</td>\n",
       "      <td>3.2313</td>\n",
       "      <td>6.1002</td>\n",
       "      <td>5.4307</td>\n",
       "      <td>4.0136</td>\n",
       "      <td>4.0454</td>\n",
       "      <td>5.0246</td>\n",
       "      <td>3.4700</td>\n",
       "      <td>6.2627</td>\n",
       "      <td>3.2420</td>\n",
       "      <td>5.1048</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TCGA-2J-AAB9-01</td>\n",
       "      <td>12.0252</td>\n",
       "      <td>11.9711</td>\n",
       "      <td>12.8020</td>\n",
       "      <td>12.7146</td>\n",
       "      <td>12.5884</td>\n",
       "      <td>12.6511</td>\n",
       "      <td>12.9294</td>\n",
       "      <td>12.8316</td>\n",
       "      <td>11.6517</td>\n",
       "      <td>...</td>\n",
       "      <td>1.9564</td>\n",
       "      <td>5.3345</td>\n",
       "      <td>4.8095</td>\n",
       "      <td>0.9935</td>\n",
       "      <td>4.0799</td>\n",
       "      <td>4.9127</td>\n",
       "      <td>2.6670</td>\n",
       "      <td>5.6153</td>\n",
       "      <td>3.3177</td>\n",
       "      <td>4.9397</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 19383 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            sample  ENSG00000198888  ENSG00000198763  ENSG00000198804  \\\n",
       "0  TCGA-2J-AAB1-01          12.5979          12.7660          13.9326   \n",
       "1  TCGA-2J-AAB4-01          11.9508          12.2991          13.7032   \n",
       "2  TCGA-2J-AAB6-01          11.0837          11.8905          12.5723   \n",
       "3  TCGA-2J-AAB8-01          10.7231          12.7880          12.1531   \n",
       "4  TCGA-2J-AAB9-01          12.0252          11.9711          12.8020   \n",
       "\n",
       "   ENSG00000198712  ENSG00000228253  ENSG00000198899  ENSG00000198938  \\\n",
       "0          13.5892          13.4221          13.6399          13.5917   \n",
       "1          13.9784          13.5229          13.5828          13.9554   \n",
       "2          12.9621          12.3837          12.5712          13.6005   \n",
       "3          13.7012          13.7530          13.4922          13.7765   \n",
       "4          12.7146          12.5884          12.6511          12.9294   \n",
       "\n",
       "   ENSG00000198840  ENSG00000212907  ...  ENSG00000160678  ENSG00000160679  \\\n",
       "0          13.4003          12.4744  ...           5.2173           5.8480   \n",
       "1          12.9003          12.4813  ...           3.6703           5.9189   \n",
       "2          11.8531          11.6337  ...           1.9149           5.5756   \n",
       "3          13.2663          12.7046  ...           3.2313           6.1002   \n",
       "4          12.8316          11.6517  ...           1.9564           5.3345   \n",
       "\n",
       "   ENSG00000143553  ENSG00000214193  ENSG00000196182  ENSG00000181817  \\\n",
       "0           4.8861           3.9127           4.1628           5.1575   \n",
       "1           4.9275           2.8381           4.5675           4.7793   \n",
       "2           5.1728           3.5174           4.1652           5.5085   \n",
       "3           5.4307           4.0136           4.0454           5.0246   \n",
       "4           4.8095           0.9935           4.0799           4.9127   \n",
       "\n",
       "   ENSG00000116885  ENSG00000116898  ENSG00000119535  ENSG00000142694  \n",
       "0           2.9338           5.9086           2.3423           4.7549  \n",
       "1           2.7615           5.8590           2.7139           4.4054  \n",
       "2           2.2663           7.0333           2.4412           5.2377  \n",
       "3           3.4700           6.2627           3.2420           5.1048  \n",
       "4           2.6670           5.6153           3.3177           4.9397  \n",
       "\n",
       "[5 rows x 19383 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%%time\n",
    "# DL data gene expression for Normal Tissue and Primary Tumors\n",
    "path = output_file_directory + \"dataframes_for_input/pancreas/\"\n",
    "\n",
    "ensembl_normal_pancreas_protcoding_expr= pandas.read_csv(path+ 'ensembl_normal_pancreas_protcoding_expr.tsv', sep = \"\\t\")\n",
    "print(ensembl_normal_pancreas_protcoding_expr.shape)\n",
    "\n",
    "directory = \"ensembl_primary_pancreas_protcoding_expr.tsv\"\n",
    "ensembl_primary_pancreas_protcoding_expr= pandas.read_csv(path+'ensembl_primary_pancreas_protcoding_expr.tsv' , sep = \"\\t\")\n",
    "print(ensembl_primary_pancreas_protcoding_expr.shape)\n",
    "ensembl_primary_pancreas_protcoding_expr.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "id": "9b0Bg43gEMRM",
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(167, 19387)\n",
      "(178, 19387)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample_Type</th>\n",
       "      <th>Primary_Site</th>\n",
       "      <th>ATG7</th>\n",
       "      <th>ATG7_1</th>\n",
       "      <th>ATG7_2</th>\n",
       "      <th>ENSG00000198888</th>\n",
       "      <th>ENSG00000198763</th>\n",
       "      <th>ENSG00000198804</th>\n",
       "      <th>ENSG00000198712</th>\n",
       "      <th>ENSG00000228253</th>\n",
       "      <th>...</th>\n",
       "      <th>ENSG00000160678</th>\n",
       "      <th>ENSG00000160679</th>\n",
       "      <th>ENSG00000143553</th>\n",
       "      <th>ENSG00000214193</th>\n",
       "      <th>ENSG00000196182</th>\n",
       "      <th>ENSG00000181817</th>\n",
       "      <th>ENSG00000116885</th>\n",
       "      <th>ENSG00000116898</th>\n",
       "      <th>ENSG00000119535</th>\n",
       "      <th>ENSG00000142694</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Normal Tissue</td>\n",
       "      <td>Pancreas</td>\n",
       "      <td>3.259</td>\n",
       "      <td>2.064</td>\n",
       "      <td>0.6239</td>\n",
       "      <td>12.5552</td>\n",
       "      <td>12.7960</td>\n",
       "      <td>13.5489</td>\n",
       "      <td>13.4342</td>\n",
       "      <td>14.4641</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0289</td>\n",
       "      <td>4.9924</td>\n",
       "      <td>4.0506</td>\n",
       "      <td>0.3907</td>\n",
       "      <td>4.6165</td>\n",
       "      <td>3.9719</td>\n",
       "      <td>2.1606</td>\n",
       "      <td>5.2028</td>\n",
       "      <td>4.8768</td>\n",
       "      <td>3.7204</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Normal Tissue</td>\n",
       "      <td>Pancreas</td>\n",
       "      <td>3.216</td>\n",
       "      <td>2.635</td>\n",
       "      <td>-1.4700</td>\n",
       "      <td>13.0496</td>\n",
       "      <td>13.6030</td>\n",
       "      <td>13.7792</td>\n",
       "      <td>14.4138</td>\n",
       "      <td>15.0967</td>\n",
       "      <td>...</td>\n",
       "      <td>0.4016</td>\n",
       "      <td>5.7918</td>\n",
       "      <td>4.8610</td>\n",
       "      <td>1.4281</td>\n",
       "      <td>4.8949</td>\n",
       "      <td>5.0960</td>\n",
       "      <td>3.0498</td>\n",
       "      <td>5.6741</td>\n",
       "      <td>1.5661</td>\n",
       "      <td>5.4858</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Normal Tissue</td>\n",
       "      <td>Pancreas</td>\n",
       "      <td>2.868</td>\n",
       "      <td>1.423</td>\n",
       "      <td>0.3346</td>\n",
       "      <td>13.1549</td>\n",
       "      <td>13.2598</td>\n",
       "      <td>13.4632</td>\n",
       "      <td>13.6357</td>\n",
       "      <td>14.4806</td>\n",
       "      <td>...</td>\n",
       "      <td>2.5263</td>\n",
       "      <td>4.9336</td>\n",
       "      <td>3.1637</td>\n",
       "      <td>-1.3548</td>\n",
       "      <td>4.2631</td>\n",
       "      <td>3.6646</td>\n",
       "      <td>1.3901</td>\n",
       "      <td>4.7814</td>\n",
       "      <td>1.1316</td>\n",
       "      <td>2.0076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Normal Tissue</td>\n",
       "      <td>Pancreas</td>\n",
       "      <td>2.830</td>\n",
       "      <td>2.095</td>\n",
       "      <td>-0.3940</td>\n",
       "      <td>12.6427</td>\n",
       "      <td>12.8969</td>\n",
       "      <td>13.5829</td>\n",
       "      <td>13.4833</td>\n",
       "      <td>13.7899</td>\n",
       "      <td>...</td>\n",
       "      <td>2.1444</td>\n",
       "      <td>5.4138</td>\n",
       "      <td>4.1986</td>\n",
       "      <td>-0.5543</td>\n",
       "      <td>4.3548</td>\n",
       "      <td>4.1194</td>\n",
       "      <td>1.8201</td>\n",
       "      <td>5.2563</td>\n",
       "      <td>4.6247</td>\n",
       "      <td>3.1212</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Normal Tissue</td>\n",
       "      <td>Pancreas</td>\n",
       "      <td>2.822</td>\n",
       "      <td>1.683</td>\n",
       "      <td>0.2998</td>\n",
       "      <td>12.9135</td>\n",
       "      <td>12.6805</td>\n",
       "      <td>12.8769</td>\n",
       "      <td>12.9432</td>\n",
       "      <td>13.9998</td>\n",
       "      <td>...</td>\n",
       "      <td>0.8883</td>\n",
       "      <td>5.0202</td>\n",
       "      <td>3.8300</td>\n",
       "      <td>-1.2481</td>\n",
       "      <td>4.3709</td>\n",
       "      <td>4.3075</td>\n",
       "      <td>1.9822</td>\n",
       "      <td>5.1065</td>\n",
       "      <td>1.3454</td>\n",
       "      <td>2.9765</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 19387 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Sample_Type Primary_Site   ATG7  ATG7_1  ATG7_2  ENSG00000198888  \\\n",
       "0  Normal Tissue     Pancreas  3.259   2.064  0.6239          12.5552   \n",
       "1  Normal Tissue     Pancreas  3.216   2.635 -1.4700          13.0496   \n",
       "2  Normal Tissue     Pancreas  2.868   1.423  0.3346          13.1549   \n",
       "3  Normal Tissue     Pancreas  2.830   2.095 -0.3940          12.6427   \n",
       "4  Normal Tissue     Pancreas  2.822   1.683  0.2998          12.9135   \n",
       "\n",
       "   ENSG00000198763  ENSG00000198804  ENSG00000198712  ENSG00000228253  ...  \\\n",
       "0          12.7960          13.5489          13.4342          14.4641  ...   \n",
       "1          13.6030          13.7792          14.4138          15.0967  ...   \n",
       "2          13.2598          13.4632          13.6357          14.4806  ...   \n",
       "3          12.8969          13.5829          13.4833          13.7899  ...   \n",
       "4          12.6805          12.8769          12.9432          13.9998  ...   \n",
       "\n",
       "   ENSG00000160678  ENSG00000160679  ENSG00000143553  ENSG00000214193  \\\n",
       "0           2.0289           4.9924           4.0506           0.3907   \n",
       "1           0.4016           5.7918           4.8610           1.4281   \n",
       "2           2.5263           4.9336           3.1637          -1.3548   \n",
       "3           2.1444           5.4138           4.1986          -0.5543   \n",
       "4           0.8883           5.0202           3.8300          -1.2481   \n",
       "\n",
       "   ENSG00000196182  ENSG00000181817  ENSG00000116885  ENSG00000116898  \\\n",
       "0           4.6165           3.9719           2.1606           5.2028   \n",
       "1           4.8949           5.0960           3.0498           5.6741   \n",
       "2           4.2631           3.6646           1.3901           4.7814   \n",
       "3           4.3548           4.1194           1.8201           5.2563   \n",
       "4           4.3709           4.3075           1.9822           5.1065   \n",
       "\n",
       "   ENSG00000119535  ENSG00000142694  \n",
       "0           4.8768           3.7204  \n",
       "1           1.5661           5.4858  \n",
       "2           1.1316           2.0076  \n",
       "3           4.6247           3.1212  \n",
       "4           1.3454           2.9765  \n",
       "\n",
       "[5 rows x 19387 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Preparation of the global dataframe with ATG7 isoform expression AND whole gene expression\n",
    "data = [[ensembl_normal_pancreas_protcoding_expr, df_ATG7_Normal], [ensembl_primary_pancreas_protcoding_expr, df_ATG7_Primary]]\n",
    "\n",
    "df_final = []\n",
    "\n",
    "for df in data:\n",
    "    #Merge the two df to have ATG7 isoform expression + whole gene expression \n",
    "  df_merged = pandas.merge(df[1], df[0], on= 'sample')\n",
    "\n",
    "  #delete the sample column and select column\n",
    "  df_merged = df_merged.drop(\"sample\", 1)\n",
    "  df_merged = df_merged.drop(df_merged.columns[5:19], axis=1)\n",
    "  \n",
    "  #save in list\n",
    "  df_final.append(df_merged)\n",
    "\n",
    "ensembl_normal_pancreas_protcoding_atg7_expr = df_final[0]\n",
    "ensembl_primary_pancreas_protcoding_atg7_expr = df_final[1]\n",
    "\n",
    "print(ensembl_normal_pancreas_protcoding_atg7_expr.shape)\n",
    "print(ensembl_primary_pancreas_protcoding_atg7_expr.shape)\n",
    "ensembl_normal_pancreas_protcoding_atg7_expr.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "#save to csv\n",
    "path = 'dataframes_for_input/pancreas/'\n",
    "ensembl_normal_pancreas_protcoding_atg7_expr.to_csv(output_file_directory+path+'ensembl_normal_pancreas_atg7_proteinGE_all.tsv', sep = \"\\t\", index=False)\n",
    "ensembl_primary_pancreas_protcoding_atg7_expr.to_csv(output_file_directory+path+'ensembl_primary_pancreas_atg7_proteinGE_all.tsv', sep = \"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "##SYMBOL NOT UPDATED"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## B) ATG7 isoforms + Whole gene expression (symbol)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# %%time\n",
    "# # DL data gene expression for Normal Tissue and Primary Tumors\n",
    "# input_file_directory_gene_expr = '/Users/kja11/OneDrive - Háskóli Íslands/PhD ATG7/0 Python Analysis/gene_expr_preparation/'\n",
    "\n",
    "# directory = input_file_directory_gene_expr + \"symbol_normal_pancreas_protcoding_expr.tsv\"\n",
    "# symbol_normal_pancreas_protcoding_expr= pandas.read_csv(directory, sep = \"\\t\")\n",
    "# print(symbol_normal_pancreas_protcoding_expr.shape)\n",
    "\n",
    "# directory = input_file_directory_gene_expr + \"symbol_primary_pancreas_protcoding_expr.tsv\"\n",
    "# symbol_primary_pancreas_protcoding= pandas.read_csv(directory, sep = \"\\t\")\n",
    "# print(symbol_primary_pancreas_protcoding.shape)\n",
    "# symbol_primary_pancreas_protcoding.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# #Preparation of the global dataframe with ATG7 isoform expression AND whole gene expression\n",
    "# data = [[symbol_normal_pancreas_protcoding, df_ATG7_Normal], [symbol_primary_pancreas_protcoding, df_ATG7_Primary]]\n",
    "\n",
    "# df_final = []\n",
    "\n",
    "# for df in data:\n",
    "#     #Merge the two df to have ATG7 isoform expression + whole gene expression \n",
    "#   df_merged = pandas.merge(df[1], df[0], on= 'sample')\n",
    "\n",
    "#   #delete the sample column and select column\n",
    "#   df_merged = df_merged.drop(\"sample\", 1)\n",
    "#   df_merged = df_merged.drop(df_merged.columns[5:19], axis=1)\n",
    "  \n",
    "#   #save in list\n",
    "#   df_final.append(df_merged)\n",
    "\n",
    "# symbol_normal_pancreas_protcoding_atg7_expr = df_final[0]\n",
    "# symbol_primary_pancreas_protcoding_atg7_expr = df_final[1]\n",
    "\n",
    "# print(symbol_normal_pancreas_protcoding_atg7_expr.shape)\n",
    "# print(symbol_primary_pancreas_protcoding_atg7_expr.shape)\n",
    "# symbol_normal_pancreas_protcoding_atg7_expr.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# #Why ATG7_x ?\n",
    "# verif = symbol_normal_pancreas_protcoding_atg7_expr[symbol_normal_pancreas_protcoding_atg7_expr.columns[pandas.Series(symbol_normal_pancreas_protcoding_atg7_expr.columns).str.startswith('ATG7')]]\n",
    "# print(verif.head(2))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# data = [symbol_normal_pancreas_protcoding_atg7_expr, symbol_primary_pancreas_protcoding_atg7_expr]\n",
    "\n",
    "# for df in data:\n",
    "#     df.drop('ATG7_y', axis = 1, inplace=True) \n",
    "#     df.rename(columns = {'ATG7_x' : 'ATG7'}, inplace=True)\n",
    "#     verif = df[df.columns[pandas.Series(df.columns).str.startswith('ATG7')]]\n",
    "#     print(verif.head(2))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#save to csv\n",
    "# symbol_normal_pancreas_protcoding_atg7_expr.to_csv('symbol_normal_pancreas_atg7_ProteinGE_all.tsv', sep = \"\\t\", index=False)\n",
    "# symbol_primary_pancreas_protcoding_atg7_expr.to_csv('symbol_primary_pancreas_atg7_ProteinGE_all.tsv', sep = \"\\t\", index=False)"
   ]
  }
 ],
 "metadata": {
  "colab": {
   "collapsed_sections": [
    "scwtwuDCqUKz",
    "fbjL51C8RltI",
    "R8fLWHXOBYJl"
   ],
   "name": "all_organs.v004.ipynb",
   "provenance": []
  },
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
